src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1195,8 +1201,29 @@ internal SingleLookupResult CheckViability(Symbol symbol, int arity, LookupOptio
             {
                 return LookupResult.Good(symbol);
             }
-        }
 
+            bool IsBadIvtSpecification()
+            {
+                // Ensures that during binding we don't ask for public key which results in attribute binding and stack overflow.
+                // If looking up attributes, don't ask for public key.
+                if ((unwrappedSymbol.DeclaredAccessibility == Accessibility.Internal ||
+                    unwrappedSymbol.DeclaredAccessibility == Accessibility.ProtectedAndInternal ||
+                    unwrappedSymbol.DeclaredAccessibility == Accessibility.ProtectedOrInternal)
+                    && !options.IsAttributeTypeLookup())
+                {
+                    foreach (ImmutableArray<byte> key in unwrappedSymbol.ContainingAssembly.GetInternalsVisibleToPublicKeys(this.Compilation.AssemblyName))
+                    {
+                        if (key.SequenceEqual(this.Compilation.Assembly.Identity.PublicKey))
+                        {
+                            return false;
+                        }
+                    }
+                    return true;
```

Comment:
This returns true even if there are no IVT attributes. The check should be that there is at least one attribute, but all of the public keys are wrong.

Reply:
line 1221 is nested within the greater Accessibility check statement, and is only called if all of the SequenceEquals have returned false / the public keys are wrong. There is a return false at line 1223, outside of the accessibility check. Are you referring to different attributes?

Reply:
DeclaredAccessibility just means what it was declared as, e.g. `internal`. No check on if it has IVT access.

Reply:
I think you can just check if the thing you're going to foreach over is empty before trying and return false if so.

Reply:
Pushed